app/src/main/java/com/hedvig/app/feature/offer/ui/changestartdate/ChangeDateBottomSheet.kt

```diff
@@ -37,94 +52,96 @@ class ChangeDateBottomSheet : BottomSheetDialogFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
-        binding.apply {
-            datePickButton.setHapticClickListener {
-                showDatePickerDialog()
+        lifecycleScope.launch {
+            changeDateBottomSheetViewModel.viewState.collect { viewState ->
+                val dateText = viewState.getFormattedDateText()
+                binding.datePickText.setText(dateText)
+
+                binding.autoSetDateSwitch.isVisible = viewState.hasSwitchableInsurer
+                binding.autoSetDateTitle.isVisible = viewState.hasSwitchableInsurer
+                if (viewState.hasSwitchableInsurer) {
+                    setupDateSwitch(viewState, dateText)
+                }
+                setupChooseDateButton(viewState)
             }
+        }
 
-            chooseDateButton.isEnabled = false
-
-            val data = arguments?.getParcelable<ChangeDateBottomSheetData>(
-                DATA
-            )
+        binding.datePickText.setHapticClickListener {
+            showDatePickerDialog()
+        }
+    }
 
-            if (data == null) {
-                e { "Programmer error: DATA not passed to ${this.javaClass.name}" }
-                return
+    private fun setupChooseDateButton(viewState: ChangeDateBottomSheetViewModel.ViewState) {
```

Comment:
What's your favorite nomenclature regarding methods that take data and align the view state with that data? I'm partial to `bind`, but I've seen `setup` being used commonly in the android world as well. `setup` always confused me a little though, like I could never distinguish between if it was a method that initialized the view (one-shot operation) or bound data to the view (to be called continuously) 🤷

Reply:
Yeah bind makes more sense :) Renamed